core/src/main/java/org/apache/accumulo/core/spi/package-info.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * This package provides a place for plugin interfaces intended for Accumulo users to implement. The
+ * types under this package should adhere to the following rules.
+ *
+ * <ul>
+ * <li>No changes should be made in a bug fix release.
+ * <li>Any changes can be made in a minor or major version. Incompatible changes should only be made
+ * if there is a benefit to users that outweighs the negative impact to users. If possible use
+ * deprecation instead of making incompatible changes.
+ * <li>All types used within this package should be declared in Accumulo's public API or under this
+ * package. This rule makes it possible to achieve the other rules. Accumulo's build uses Apilyzer
+ * to check this rule.
```

Comment:
Is this already being done?  I only see the addition of iterators to the apilyzer in this PR.

Reply:
Yeah, it does exists.  I took a look at it and saw it could be simplified and I just did that.